s3api/controllers/base.go

```diff
@@ -800,15 +816,21 @@ func SendResponse(ctx *fiber.Ctx, err error) error {
 
 	utils.LogCtxDetails(ctx, []byte{})
 
+	if l.Logger != nil {
+		l.Logger.Log(ctx, nil, nil, l.Meta)
+	}
+
 	// https://github.com/gofiber/fiber/issues/2080
 	// ctx.SendStatus() sets incorrect content length on HEAD request
 	ctx.Status(http.StatusOK)
 	return nil
 }
 
-func SendXMLResponse(ctx *fiber.Ctx, resp any, err error) error {
+func SendXMLResponse(ctx *fiber.Ctx, resp any, err error, l *LogOptions) error {
 	if err != nil {
-		fmt.Println(err)
+		if l.Logger != nil {
+			l.Logger.Log(ctx, err, nil, l.Meta)
+		}
```

Comment:
same here, move up one line, remove success case logger

Reply:
Here we need the logger separated for error and success cases to get the exact body size

Reply:
oh i see, yes, thats fine